cmd/geth/config.go

```diff
@@ -294,7 +295,78 @@ func makeFullNode(ctx *cli.Context) (*node.Node, ethapi.Backend) {
 		// rather than working correctly until then and freezing at activation until TBC is caught up.
 		// Would rather have node delay full startup to fully sync TBC to required genesis height
 		// than startup without full information and later freeze at hVM Phase 0 activation time.
+
 		for {
+
+			// wait until we have a high degree of confidence that we have
+			// chain up to the hvm genesis block indexed
+			for {
+				select {
+				case <-time.After(5 * time.Second):
+					bh, _, err := vm.TBCFullNode.BlockHeaderBest(ctx.Context)
+					if err != nil {
+						log.Crit(fmt.Sprintf("error getting block header best: %s", err))
+					}
+
+					log.Info(fmt.Sprintf("the current best block height is %d", bh))
+
+					if bh < genesisHeight {
+						continue
+					}
+
+					ghh := genesisHeader.BlockHash()
+					// we're above genesis height, but are we on the best chain with genesis?
+					_, height, err := vm.TBCFullNode.BlockHeaderByHash(ctx.Context, &ghh)
+					if err != nil {
+						log.Crit(fmt.Sprintf("could not get block by hash: %s", err))
+					}
+
+					if height != genesisHeight {
+						log.Crit(fmt.Sprintf("mismatched genesis heights %d != %d", height, genesisHeight))
+					}
+
+					break
+				case <-ctx.Context.Done():
+					return nil, nil
+				}
+				break
+			}
+
+			log.Info("done getting hvm genesis header")
+
+			// walk back to btc genesis from hvm genesis and ensure we have
+			// all of the full blocks
+			tmp := genesisHash
+			for {
```

Comment:
I am currently testing this, to make sure it doesn't take too long just to walk back the chain and ensure we have full blocks for each before we index.